atc/creds/vault/api_client.go

```diff
@@ -218,6 +220,23 @@ func (ac *APIClient) baseClient() (*vaultapi.Client, error) {
 		return nil, err
 	}
 
+	config.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		retryable, retryErr := retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+		if retryErr != nil || retryable {
+			return retryable, retryErr
+		}
+		// We also want to retry when hitting Vault rate limit.
+		if resp.StatusCode == 429 {
```

Comment:
I could only find https://www.vaultproject.io/api-docs/system/health where it says 

`[429](https://www.vaultproject.io/api-docs/system/health#429) if unsealed and standby`

is 429 a status code for rate limit only ? If not we might want to add some extra check to make sure the scope of this change.

Reply:
Good catch. Per https://github.com/hashicorp/vault/blob/main/api/response.go#L32, we can further check request path.